guice/src/main/java/com/flipkart/gjex/grpc/service/ApiServer.java

```diff
@@ -40,19 +44,39 @@
 @Named("APIServer")
 public class ApiServer extends AbstractService implements Logging {
 
-    private final Server apiServer;
-    private final ResourceRegistrar resourceRegistrar;
-	private List<ResourceConfig> resourceConfigs = new LinkedList<ResourceConfig>();
+	private final Server apiServer;
+	private final ResourceRegistrar resourceRegistrar;
+	private final ServletContextHandler context;
+	private final Filter accessLogFilter;
+	private List<ResourceConfig> resourceConfigs = new ArrayList<>();
 
 	@Inject
-	public ApiServer(@Named("APIJettyServer") Server apiServer, ResourceRegistrar resourceRegistrar) {
+	public ApiServer(@Named("APIJettyServer") Server apiServer,
+									 @Named("ApiServletContext")ServletContextHandler context,
+									 @Named("HttpAccessLogFilter")Filter accessLogFilter,
+									 ResourceRegistrar resourceRegistrar) {
 		this.apiServer = apiServer;
+		this.context = context;
+		this.accessLogFilter = accessLogFilter;
 		this.resourceRegistrar = resourceRegistrar;
 	}
 	
 	public void registerResources(List<ResourceConfig> resourceConfigs) {
 		this.resourceConfigs.addAll(resourceConfigs);
 	}
+
+	public void registerHttpFilters(List<HttpFilterParams> httpFilterParamsList,
+																	boolean enableAccessLogs){
+		if (enableAccessLogs) {
+			context.addFilter(new FilterHolder(accessLogFilter), "/*" ,
+					EnumSet.of(DispatcherType.REQUEST));
+		}
```

Comment:
we should make extract this out to a private separate function

Reply:
done